clients/ui/frontend/src/app/pages/modelCatalog/components/ModelCatalogCard.tsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import {
+  Button,
+  Card,
+  CardBody,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+  Flex,
+  FlexItem,
+  Icon,
+  Label,
+  Skeleton,
+  Split,
+  SplitItem,
+  Stack,
+  StackItem,
+  Truncate,
+} from '@patternfly/react-core';
+import { TagIcon } from '@patternfly/react-icons';
+import { useNavigate } from 'react-router-dom';
+import { ModelCatalogItem } from '~/app/modelCatalogTypes';
+import ModelCatalogLabels from './ModelCatalogLabels';
+
+type ModelCatalogCardProps = {
+  model: ModelCatalogItem;
+  source: string;
+  truncate?: boolean;
+  onSelect?: (model: ModelCatalogItem) => void;
+};
+
+const ModelCatalogCard: React.FC<ModelCatalogCardProps> = ({
+  model,
+  source,
+  truncate = false,
+  onSelect,
+}) => {
+  const navigate = useNavigate();
+
+  // Extract version from tags (assuming version tag matches semantic versioning pattern)
+  const versionTag = model.tags?.find((tag) => /^\d+\.\d+\.\d+$/.test(tag));
```

Comment:
Not sure if this is used somewhere else, but I would extract this as a util finction, it makes it more readable.